readers/src/main/scala/com/salesforce/op/readers/JoinedDataReader.scala

```diff
@@ -346,6 +346,8 @@ private[op] class JoinedAggregateDataReader[T, U]
 }
 
 
+// TODO: UserDefinedAggregateFunction is now deprecated in favor of Aggregator,
```

Comment:
@tovbinm / @leahmcguire , I could use some pointers/help here. Since this is largely unused anyway, I'm even considering dropping the join functionality.

Reply:
Good idea. I would also use this as an opportunity to get rid of other unused code, e.g cli with codegen.

Reply:
@leahmcguire ^^^

Besides the immediate problems of upgrading the UDAF approach, we're trying to reduce the burden of maintenance a bit here too . And since we're all in on Spark SQL on our end, we have no need for it anymore.

At this point, it's still open for discussion/contributions, though.

Reply:
@emitc2h , we could look into hacking it like we did here: https://github.com/salesforce/TransmogrifAI/pull/550/commits/a946ffb01e6eb7a1acd5636774ed19967f9479f4